core/trace/src/main/scala/org/typelevel/otel4s/trace/SpanBuilder.scala

```diff
@@ -112,6 +113,15 @@ trait SpanBuilder[F[_]] {
   def withParent(parent: SpanContext): SpanBuilder[F]
 
   def build: SpanOps[F]
+
+  /** Modify the context `F` using an implicit [[KindTransformer]] from `F` to
+    * `G`.
+    */
+  def mapK[G[_]: MonadCancelThrow](implicit
```

Comment:
Can we lift `MonadCancelThrow[G]` similarly to https://github.com/typelevel/otel4s/pull/264/files#diff-36bdd6d74392dca0ddc0d9c6312f06c778e77a49e87fbc5c6fc966c936171f1cR248?

Reply:
I can't figure out a way to do it without `G ~> F`; if you can, feel free to lmk or even add it yourself

Reply:
No, we cannot do it in this case.